packages/shared/components/inputs/TokenAmountInput.svelte

```diff
@@ -1,28 +1,74 @@
 <script lang="ts">
-    import { formatCurrency } from '@core/i18n'
+    import { formatCurrency, getDecimalSeparator } from '@core/i18n'
     import { getMaxDecimalsFromTokenMetadata } from '@core/token/utils'
     import { IAsset, convertToRawAmount, formatTokenAmountDefault, visibleSelectedAccountAssets } from '@core/wallet'
     import { AmountInput, FontWeight, InputContainer, Text } from '@ui'
     import { getMarketAmountFromAssetValue } from '@core/market/utils'
     import { validateTokenAmount } from '@core/wallet/utils/validateTokenAmount'
+    import { getBaseToken } from '@core/profile'
 
     export let asset: IAsset | undefined = $visibleSelectedAccountAssets?.baseCoin
     export let rawAmount: string | undefined = undefined
     export let unit: string | undefined = undefined
+    export let availableBalance: number
     export let inputtedAmount: string | undefined =
         rawAmount && asset?.metadata
             ? formatTokenAmountDefault(Number(rawAmount), asset.metadata, unit, false)
             : undefined
 
     let amountInputElement: HTMLInputElement
     let error: string
+    let inputLength: number = 0
+    let fontSize: number = 64
+    let maxLength: number = 0
```

Comment:
In cases where we initialize a variable as a number, it's redundant to type them. Typing is only required for exported variables, so that the parent components also throw errors if the type is not correctly adhered to.

The elements on l.19 & l.20 are `undefined` at initialization, so maybe that type should be added as well?